src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts

```diff
@@ -120,6 +122,64 @@ class MLSConversationVerificationStateHandler {
     await this.checkAllConversationsVerificationState();
   };
 
+  private checkUserHandle = (identity: WireIdentity, user: User): boolean => {
+    // WireIdentity handle format is "{scheme}%40{username}@{domain}"
+    // Example: wireapp://%40hans.wurst@elna.wire.link
+    const {handle: identityHandle} = identity;
+    // We only want to check the username part of the handle
+    const regex = /.*%40([^@]+)@[^@]+/;
+    const match = identityHandle.match(regex);
+    if (!match) {
+      return false;
+    }
+
+    const {username: userHandle} = user;
+
+    return match[1] === userHandle();
+  };
+
+  private checkAllUserCredentialsInConversation = async (
+    conversation: MLSCapableConversation,
+  ): Promise<{
+    userVerificationState: UserVerificationState;
+    userIdentities: Map<string, WireIdentity[]>;
+    problematicUserIds: Set<string>;
+  }> => {
+    const userIdentities = await getAllGroupUsersIdentities(conversation.groupId);
+    const processedUserIds: string[] = [];
+    const problematicUserIds: Set<string> = new Set();
+
+    for (const [userId, identities] of userIdentities.entries()) {
+      if (processedUserIds.includes(userId)) {
+        continue;
+      }
+      processedUserIds.push(userId);
+
+      const user = await waitFor(() => conversation.allUserEntities().find(user => user.qualifiedId.id === userId));
```

Comment:
We have a race condition here in certain circumstances. In some cases the users are locally not updated when we get the epochChange event.

Reply:
Please add a comment in the code explaining the race condition. This should not be left uncommented. 

Please explain the exact reason why there is a race condition